scripts/sct_deepseg.py

```diff
@@ -0,0 +1,93 @@
+#!/usr/bin/env python
+# -*- coding: utf-8
+# This command-line tool is the interface for the deepseg API that performs segmentation using deep learning from the
+# ivadomed package.
+
+from __future__ import absolute_import
+
+import sys
+import os
+import argparse
+
+from spinalcordtoolbox.utils import Metavar, SmartFormatter
+from spinalcordtoolbox.deepseg.core import segment_nifti
+
+from sct_utils import init_sct
+
+
+def get_parser():
+    parser = argparse.ArgumentParser(
+        description="Segmentation using deep learning.",
+        add_help=None,
+        formatter_class=SmartFormatter,
+        prog=os.path.basename(__file__).strip(".py"))
```

Comment:
`argparse` already has this logic built in, if you let it:
```suggestion
        formatter_class=SmartFormatter)
```

It uses sys.argv[0] instead of __file__, but IMO that is the more Unix-canonical way, because that matches how the program is actually installed and known to the system, instead of what the particular filename this code is in is.

Reply:
hum, if i implement the suggestion, the removal of `add_help=None,` generates the following error:
~~~
$ sct_deepseg

--
Spinal Cord Toolbox (git-jca/2626-deepseg-021f205c1009266db7ec3ca553f322a9fa6c7dcd*)

Traceback (most recent call last):
  File "/Users/julien/code/sct/scripts/sct_deepseg.py", line 79, in <module>
    main()
  File "/Users/julien/code/sct/scripts/sct_deepseg.py", line 67, in main
    parser = get_parser()
  File "/Users/julien/code/sct/scripts/sct_deepseg.py", line 59, in get_parser
    help="Show this help message and exit")
  File "/Users/julien/code/sct/python/envs/venv_sct/lib/python3.6/argparse.py", line 1352, in add_argument
    return self._add_action(action)
  File "/Users/julien/code/sct/python/envs/venv_sct/lib/python3.6/argparse.py", line 1556, in _add_action
    action = super(_ArgumentGroup, self)._add_action(action)
  File "/Users/julien/code/sct/python/envs/venv_sct/lib/python3.6/argparse.py", line 1366, in _add_action
    self._check_conflict(action)
  File "/Users/julien/code/sct/python/envs/venv_sct/lib/python3.6/argparse.py", line 1505, in _check_conflict
    conflict_handler(action, confl_optionals)
  File "/Users/julien/code/sct/python/envs/venv_sct/lib/python3.6/argparse.py", line 1514, in _handle_conflict_error
    raise ArgumentError(action, message % conflict_string)
argparse.ArgumentError: argument -h/--help: conflicting option strings: -h, --help
~~~

If I remove `prog=os.path.basename(__file__).strip(".py"))`, the usage displays the .py extension here:
~~~
usage: sct_deepseg.py -i <file> [-m {cord-t2star,uqueensland-mice-sc,uqueensland-mice-gm}] [-o <class 'str'>] [-v {0,1}] [-h]
~~~
I understand the argument about the Unix-canonical way, but I've had users who would write "sct_XXX.py" (with the extension), and then do all sorts of things to get it work (e.g. write full path to the SCT_DIR/script), which obviously didn't work because they need to actually run the launcher (alias to the proper venv). So in brief: i'd like to minimize confusion for users and not display the extension because this is how they should call the function.

Reply:
Have you tried renaming the file without the extension? So long as "#!/usr/bin/env python" is the first line it will still work.

Reply:
yes, but my IDE (Pycharm) freaks out if a python file does not have the .py extension 😐

Reply:
Ah I see. No worries then. This works as well.